File: 2013/submissions/Wesley Taylor/1/ruby/word_frequency.rb
File.open("freqs.txt", 'w'){|f| (IO.read ARGV[0]).gsub(/[^A-Za-z\s]/, "").split().reduce(Hash.new{|i|0}){|res,w| res[w.downcase]+=1;res}.sort_by {|k,v| -v}.each do |k,v|  f.write("#{v} #{k}\n") end}

File: 2013/submissions/Wesley Taylor/1/perl/word_frequency.pl
$f = shift;
local $/=undef;
open $i, '<', $f;
open $o, '>', 'freqs.txt';
$l = <$i>;
$l =~ s/[^A-Za-z\s]//g;
foreach $s (split /\s+/, $l) {
    $c{lc($s)}++;
}

foreach $s (sort { $c{$b} <=> $c{$a} } keys %c) {
    printf $o "%s %s\n", $c{$s},  $s;
}
